lib/features/give/bloc/give/give_bloc.dart

```diff
@@ -647,4 +656,60 @@ class GiveBloc extends Bloc<GiveEvent, GiveState> {
   ) {
     emit(const GiveState());
   }
+
+  /// Tracks donations to organizations and automatically adds an organization to favorites
+  /// after a user gives to the same organization twice in a row (consecutive donations)
+  Future<void> _handleAutoFavorites(String namespace, String userGUID) async {
+    try {
+      final prefs = await SharedPreferences.getInstance();
+
+      // Get previous donation organization
+      final previousDonationKey = '${Util.previousDonationOrgKey}_$userGUID';
+      final previousDonationOrg = prefs.getString(previousDonationKey) ?? '';
+
+      // Check if this is a consecutive donation to the same organization
+      final isConsecutiveDonation = namespace == previousDonationOrg;
+
+      // If this is the second consecutive donation to the same organization, add it to favorites
+      if (isConsecutiveDonation && previousDonationOrg.isNotEmpty) {
+        // Get favorited organizations
+        final favoritesKey = '${Util.favoritedOrganisationsKey}$userGUID';
```

Comment:
Does it really add something to add the User? I think on logout + login with different user all localstorage keys are cleared

Reply:
Favorites is already live so if i would stop doing it now it would reset for current users.

Reply:
ah okay